mbedit_prog.c: operation on ‘*(ping[i].beamflag + (sizetype)j)’ may be undefined [-Wsequence-point]

With 

```gcc --version
gcc (Debian 8.3.0-6) 8.3.0
```

I see:

```
gcc -DHAVE_CONFIG_H -I. -I../../src/mbio  -I../../src/mbio -I../../src/mbaux -I../../src/gsf  -I/usr/include    -g -O2 -Wno-format -Wno-stringop-truncation -Wno-stringop-overflow -Wall -Wextra -Wno-sign-compare -Wno-maybe-uninitialized -Wno-format-overflow -MT mbedit_prog.o -MD -MP -MF $depbase.Tpo -c -o mbedit_prog.o mbedit_prog.c &&\
mv -f $depbase.Tpo $depbase.Po
mbedit_prog.c: In function ‘mbedit_action_mouse_restore’:
mbedit_prog.c:1847:28: warning: operation on ‘*(ping[i].beamflag + (sizetype)j)’ may be undefined [-Wsequence-point]
        ping[i].beamflag[j] = ping[i].beamflag[j] = mb_beam_set_flag_none(ping[i].beamflag[j]);
mbedit_prog.c: In function ‘mbedit_filter_ping’:
mbedit_prog.c:3343:32: warning: operation on ‘*(ping[iping].beamflag + (sizetype)j)’ may be undefined [-Wsequence-point]
        ping[iping].beamflag[j] = ping[iping].beamflag[j] = mb_beam_set_flag_filter2(ping[iping].beamflag[j]);
mbedit_prog.c:3364:32: warning: operation on ‘*(ping[iping].beamflag + (sizetype)j)’ may be undefined [-Wsequence-point]
        ping[iping].beamflag[j] = ping[iping].beamflag[j] = mb_beam_set_flag_filter2(ping[iping].beamflag[j]);
mbedit_prog.c:3391:33: warning: operation on ‘*(ping[iping].beamflag + (sizetype)j)’ may be undefined [-Wsequence-point]
         ping[iping].beamflag[j] = ping[iping].beamflag[j] =
mbedit_prog.c:3415:33: warning: operation on ‘*(ping[iping].beamflag + (sizetype)j)’ may be undefined [-Wsequence-point]
         ping[iping].beamflag[j] = ping[iping].beamflag[j] =
```

e.g.  What is this supposed to be?  It looks like `a = a = mb_beam_set_flag_none(a)`

```c
if (!mb_beam_ok(ping[i].beamflag[j]) && !mb_beam_check_flag_unusable(ping[i].beamflag[j]))
    ping[i].beamflag[j] = ping[i].beamflag[j] = mb_beam_set_flag_none(ping[i].beamflag[j]);
```

--- a/src/mbedit/mbedit_prog.c
+++ b/src/mbedit/mbedit_prog.c
@@ -1844,7 +1844,7 @@ int mbedit_action_mouse_restore(int x_loc, int y_loc, int plwd, int exgr, int xn
 
 						/* reset the beam value */
 						if (!mb_beam_ok(ping[i].beamflag[j]) && !mb_beam_check_flag_unusable(ping[i].beamflag[j]))
-							ping[i].beamflag[j] = ping[i].beamflag[j] = mb_beam_set_flag_none(ping[i].beamflag[j]);
+							ping[i].beamflag[j] = mb_beam_set_flag_none(ping[i].beamflag[j]);
 						if (verbose >= 1) {
 							fprintf(stderr, "\nping: %d beam:%d depth:%10.3f ", i, j, ping[i].bath[j]);
 							fprintf(stderr, " flagged\n");
@@ -3340,7 +3340,7 @@ int mbedit_filter_ping(int iping) {
 						/* reset the beam value - if beam was originally flagged
 						    then reset to the original flag value */
 						if (mb_beam_ok(ping[iping].beamflagorg[j]))
-							ping[iping].beamflag[j] = ping[iping].beamflag[j] = mb_beam_set_flag_filter2(ping[iping].beamflag[j]);
+							ping[iping].beamflag[j] = mb_beam_set_flag_filter2(ping[iping].beamflag[j]);
 						else
 							ping[iping].beamflag[j] = ping[iping].beamflagorg[j];
 						if (verbose >= 1) {
@@ -3361,7 +3361,7 @@ int mbedit_filter_ping(int iping) {
 						/* reset the beam value - if beam was originally flagged
 						    then reset to the original flag value */
 						if (mb_beam_ok(ping[iping].beamflagorg[j]))
-							ping[iping].beamflag[j] = ping[iping].beamflag[j] = mb_beam_set_flag_filter2(ping[iping].beamflag[j]);
+							ping[iping].beamflag[j] = mb_beam_set_flag_filter2(ping[iping].beamflag[j]);
 						else
 							ping[iping].beamflag[j] = ping[iping].beamflagorg[j];
 						if (verbose >= 1) {
@@ -3388,7 +3388,7 @@ int mbedit_filter_ping(int iping) {
 							/* reset the beam value - if beam was originally flagged
 							    then reset to the original flag value */
 							if (mb_beam_ok(ping[iping].beamflagorg[j]))
-								ping[iping].beamflag[j] = ping[iping].beamflag[j] =
+								ping[iping].beamflag[j] =
 								    mb_beam_set_flag_filter2(ping[iping].beamflag[j]);
 							else
 								ping[iping].beamflag[j] = ping[iping].beamflagorg[j];
@@ -3412,7 +3412,7 @@ int mbedit_filter_ping(int iping) {
 							/* reset the beam value - if beam was originally flagged
 							    then reset to the original flag value */
 							if (mb_beam_ok(ping[iping].beamflagorg[j]))
-								ping[iping].beamflag[j] = ping[iping].beamflag[j] =
+								ping[iping].beamflag[j] =
 								    mb_beam_set_flag_filter2(ping[iping].beamflag[j]);
 							else
 								ping[iping].beamflag[j] = ping[iping].beamflagorg[j];

